### PR TITLE
My Site Dashboard: refine posts card

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/17873

https://user-images.githubusercontent.com/7040243/154535166-a37fa581-f9f0-4155-889c-429f54b53192.mp4

This PR:

* Adds the "next post" prompt then the user don't have drafts or scheduled posts
* Opens the editor when the prompt is tapped
* Open drafts or scheduled post list when tapping the header of drafts/scheduled card
* Show scheduled date for scheduled posts (instead of content)

Analytics and improvements will be implemented in a subsequent task.

### To test

Grab a coffee ☕ or tea 🍵! Then, make sure you have a blog with only scheduled or draft posts, a blog with both and a blog with no posts at all.

#### No posts

1. Open the app
2. Tap Dashboard
3. Make sure the prompt appears asking you to create your **first post** ✅
4. Tap the prompt
5. The editor should open ✅ 
6. Create a draft post and save (X -> Save draft)
7. The card now should be "Work on a draft post" ✅ 
8. Go to Posts > Draft
9. Trash the draft you just created
10. Go back to the dashboard
11. The prompt should appear again for your **first post** ✅ 
12. Tap it
13. Create and publish a post
14. The prompt should appear again asking you to write your **next post** ✅

#### Drafts

1. Switch to a blog with drafts
2. Check that drafts are shown  ✅
3. Tap the header
4. The post list should appear in the "drafts" tab  ✅

#### Scheduled

1. Switch to a blog with scheduled
2. Check that scheduled posts are shown with a date ✅
3. Tap the header
4. The post list should appear in the "drafts" tab  ✅

#### Drafts and schedule

1. Open a blog with scheduled and draft posts
2. Check that drafts appear first and then scheduled posts ✅

Ps.: there are a few constraint warnings being triggered for this case. I'm not sure what's causing them but apparently there's no layout issue, I'll add as a future task.

#### Offline

1. Go offline
2. Open a blog with no posts
3. Check that when you create a post it will show your draft  ✅

Ps.: when showing only scheduled posts, this user-case will not work. For now, I'll consider that as nice to have because taking into account all posts combination would probably be a huge effort for edge cases.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by leandroalonso** [comment]
@pachlava this task has some feature changes, but I'd like to ask you to wait to test these. The next task I'm going to tackle will add some important improvements to this flow, so it would be good for you to test once those are done.

Otherwise, something you test here might end up broken in my future task.

**Comment by pachlava** [comment]
Thanks for the heads-up @leandroalonso! I'll wait for the next PR :-)

**Review by momo-ozawa — COMMENTED** [review]
@leandroalonso When I switch to the dashboard for a site that has both draft posts and scheduled posts, I'm seeing two "Create your first post" cards:


Simulator Screen Shot - iPhone 13 - 2022-02-18 at 09 43 57 | Simulator Screen Shot - iPhone 13 - 2022-02-18 at 09 27 50 | Simulator Screen Shot - iPhone 13 - 2022-02-18 at 09 47 47 
-- | -- | --

**Comment by momo-ozawa** [comment]
Also, some times I'm seeing the "Create your first post" for scheduled posts. (This was after doing a fresh install)

 | 
-- | --

**Comment by leandroalonso** [comment]
@momo-ozawa in order to reproduce the issue you were seeing you'd need to:

1. Do a clean install or logout
2. Login
3. Tap Dashboard
4. Switch sites
5. Notice that the first load for the other sites always fail

This was a rabbit hole, but I managed to understand what was going on:

1. We rely on `blog.userID` to fetch posts and to show them
2. However, this is a property that is only available after `BlogService.syncBlogAndAllMetadata` was called on the Site Menu (this is why for the first site it normally worked)
3. For the rest of the sites, this property wasn't there until you navigated to "Site Menu", so we weren't syncing correctly or displaying it

In order to fix it, I added some logic to sync authors before showing posts.

**Review by momo-ozawa — COMMENTED** [review]
@leandroalonso The test cases for No Posts, Drafts, Scheduled, Drafts + Scheduled work as expected. However, I found some new issues for the Offline test case.

1. When I turn off the wifi then switch to a site with no posts, the card title is "Work on a draft post" and "Unable to load posts right now" is displayed (see first screenshot)
2. Then if I switch to the site menu then back to the dashboard, I see "Create your first post" superimposed on "Unable to load posts right now" (see second screenshot)

 |  
-- | --

**Comment by leandroalonso** [comment]
@momo-ozawa fixed on 1f728db6df0caa7ade8145bfeeddda8252e69dc3

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsCardCell.swift [hunk 1]
```diff
@@ -29,60 +32,62 @@ class DashboardPostsCardCell: UICollectionViewCell, Reusable, BlogDashboardCardC
...
-        }
+        stackView.removeAllSubviews()
 
-        draftPostsViewController = nil
-        scheduledPostsViewController = nil
+        childVcs?.forEach { remove(child: $0) }
     }
 
-    private func embed(child childViewController: UIViewController, to viewController: UIViewController, with title: String) {
+    private func showCard(forBlog blog: Blog, status: BasePost.Status, to viewController: UIViewController, hasPublishedPosts: Bool, hiddenHeader: Bool = false) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsCardCell.swift:65]: [nit]
```suggestion
    private func showCardForBlog(_ blog: Blog, status: BasePost.Status, in viewController: UIViewController, hasPublishedPosts: Bool, hidesHeader: Bool = false) {
```

**Review by momo-ozawa — APPROVED** [review]
Works as described! Left 1 minor suggestion

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsCardCell.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsCardCell.swift:65]: I'll address this in an upcoming PR that changes this function!
